.buildkite/steps/extract-agent-version-metadata.sh

```diff
@@ -1,7 +1,7 @@
 #!/bin/bash
 set -euo pipefail
 
-agent_version=$(awk -F\" '/var baseVersion string = "/ {print $2}' agent/version.go)
+agent_version=$(cat agent/VERSION | xargs) # xargs to trim trailing newlines
```

Comment:
Nit: I don't think `xargs` is necessary:

> The shell shall expand the command substitution by executing command in a subshell environment (see [Shell Execution Environment](https://pubs.opengroup.org/onlinepubs/009695399/utilities/xcu_chap02.html#tag_02_12)) and replacing the command substitution (the text of command plus the enclosing "$()" or backquotes) with the standard output of the command, removing sequences of one or more <newline>s at the end of the substitution. Embedded <newline>s before the end of the output shall not be removed; however, they may be treated as field delimiters and eliminated during field splitting, depending on the value of IFS and quoting that is in effect.

Source: https://pubs.opengroup.org/onlinepubs/009695399/utilities/xcu_chap02.html#tag_02_06_03

Experimental evidence:
```
🕙 16:37:47 ❯ docker run --rm -it bash:latest
bash-5.2# printf $(printf "hello\n")
hellobash-5.2# printf $(printf "\nhello\n")
hellobash-5.2# printf "$(printf "\nhello\n")"

hellobash-5.2#
```

Reply:
TIL! Rebased.